prime-dotnet-webapi/Controllers/SitesController.cs

```diff
@@ -385,5 +388,65 @@ public async Task<ActionResult> sendRemoteUsersEmail(int siteId)
             return NoContent();
         }
 
+        // POST: api/Sites/5/approval
+        /// <summary>
+        /// Approved a site, setting it's approval date
+        /// </summary>
+        /// <param name="siteId"></param>
+        [HttpPost("{siteId}/approval", Name = nameof(ApproveSite))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [Authorize(Policy = AuthConstants.ADMIN_POLICY)]
+        public async Task<ActionResult<Site>> ApproveSite(int siteId)
+        {
+            var site = await _siteService.GetSiteAsync(siteId);
+            if (site == null)
+            {
+                return NotFound(ApiResponse.Message($"Site not found with id {siteId}"));
+            }
+
+            var result = await _siteService.ApproveSite(siteId);
+            return Ok(ApiResponse.Result(result));
+        }
+
+        // POST: api/Sites/5/site-registration-notes
+        /// <summary>
+        /// Creates a new site registration note on an enrollee.
+        /// </summary>
+        /// <param name="siteId"></param>
+        /// <param name="note"></param>
+        [HttpPost("{siteId}/site-registration-notes", Name = nameof(CreateSiteRegistrationNote))]
+        [Authorize(Policy = AuthConstants.ADMIN_POLICY)]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<AdjudicatorNote>), StatusCodes.Status201Created)]
+        public async Task<ActionResult<AdjudicatorNote>> CreateSiteRegistrationNote(int siteId, FromBodyText note)
+        {
+            var site = await _siteService.GetSiteAsync(siteId);
+            if (site == null)
+            {
+                return NotFound(ApiResponse.Message($"Site not found with id {siteId}"));
+            }
+            if (string.IsNullOrWhiteSpace(note))
+            {
+                this.ModelState.AddModelError("note", "site registration notes can't be null or empty.");
+                return BadRequest(ApiResponse.BadRequest(this.ModelState));
+            }
+
+            var admin = await _adminService.GetAdminAsync(User.GetPrimeUserId());
+
+            var createdSiteRegistrationNote = await _siteService.CreateSiteRegistrationNoteAsync(siteId, note, admin.Id);
+
+            return CreatedAtAction(
+                nameof(CreateSiteRegistrationNote),
```

Comment:
if there is no location where we can get the note, this should just return 200 OK. This is creating a redirect url to get the note that points back at this action to create a note.